src/openrct2/core/String.cpp

```diff
@@ -103,9 +113,28 @@ namespace String
 #else
         icu::UnicodeString str = icu::UnicodeString::fromUTF8(std::string(src));
 
+// Which constructor to use depends on the size of wchar_t...
+// UTF-32 is the default on most POSIX systems; Windows uses UTF-16.
+// Unfortunately, we'll have to help the compiler here.
+#if U_SIZEOF_WCHAR_T==4
+        size_t length = (size_t) str.length();
+        const wchar_t* buffer = (wchar_t*) malloc(length * sizeof(wchar_t));
+        if (buffer == nullptr)
+            throw std::bad_alloc();
+
+        UErrorCode status = U_ZERO_ERROR;
+        str.toUTF32((UChar32*) buffer, str.length(), status);
+
+        std::wstring result = buffer;
+
+#elif U_SIZEOF_WCHAR_T==2
         const char16_t* buffer = str.getBuffer();
         std::wstring result = (wchar_t*) buffer;
 
+#else
+#error Unsupported U_SIZEOF_WCHAR_T size
+#endif
```

Comment:
I'm really dissatisfied with this block. We could just assume `U_SIZEOF_WCHAR_T==4` and be done with it, but if we ever want to use this code path on Windows, we'll need something like this…